config.m4

```diff
@@ -5,6 +5,13 @@ PHP_ARG_ENABLE([mongodb],
                                [Enable MongoDB support])])
 
 if test "$PHP_MONGODB" != "no"; then
+  dnl Enable C99 (required for libmongoc 1.24+)
+  AC_PROG_CC_C99
+
+  if test "$ac_cv_prog_cc_c99" = no; then
+    AC_MSG_ERROR([Compiler does not support C99])
+  fi
```

Comment:
Similar to the approach taken in https://github.com/php/php-src/pull/6317.

Note: C99 was added to php-src in https://github.com/php/php-src/commit/b51a99ae358b3295de272bb3c3edb6913eeb0fd4